### ISSUE TITLE
"Failed to set permissions on the temporary files Ansible needs to create..."

### ISSUE THREAD

**Issue by djowett** [plone/ansible.plone_server#86]
Occurs on e.g. "plone.plone_server : Bootstrap buildout" when running Ansible 2.1 without being root.
(this is the case in Plone 4.3, but it looks to me like it would just give the same error on a different task in Plone 5) 

Full message is:

fatal: [peacock]: FAILED! => {"failed": true, "msg": "Failed to set permissions on the temporary files Ansible needs to create when becoming an unprivileged user. For information on working around this, see https://docs.ansible.com/ansible/become.html#becoming-an-unprivileged-user"}

Not an error in this role, but could we advise a course of action?

Basically the options are:
- use pipelining
- Install filesystem acl support on the managed host
- always run as root
- don't use become in our tasks (not suitable for us)
- (potentially insecure) override Ansible's check by setting `allow_world_readable_tmpfiles` in `ansible.cfg`

**Comment by djowett** [comment]
At the moment my Ansible clients are all running Ubuntu so pipelining is easiest for me (and apparently gives a speed increase too).  Ubuntu has requiretty off by default, so no need to worry about implications of tweaking security settings under `/etc/sudoers`.

For that, add `ansible.cfg` in playbook directory containing:

    [ssh_connection]
    pipelining=True

**Comment by smcmahon** [comment]
I'm not sure what the status of this issue is. Is this a bug report, a feature request or a documentation suggestion?

**Comment by djowett** [comment]
It looks like 7646df53d and 4e4295a resolve this with addition of ansible_ssh_pipelining - they describe the same issue in any case.

**Comment by enisozgen** [comment]
After doing what @djowett wrote

I additionally installed `sudo apt-get install acl` than everything worked better.

ref: https://stackoverflow.com/a/56379678/3439922
